### PR TITLE
ServiceBusResourceBuilder

### PR DESCRIPTION
**Pull request by dstsyst**
## Description

Show how we can use the "new" Integration Test (CI) environment to support developers in integration testing using Azure resources. See IntegrationTestConfiguration and related.

Show how we can remove dependency to Squadron for integration tests depending on a Azure Service Bus. See ServiceBusResourceProvider and related.

## References

* https://app.zenhub.com/workspaces/mighty-ducks-60ae1565be2dfe0017ff66b2/issues/energinet-datahub/mightyducks/244
* https://app.zenhub.com/workspaces/mighty-ducks-60ae1565be2dfe0017ff66b2/issues/energinet-datahub/mightyducks/245

### CODE REVIEW

File: source/TestCommon/documents/functionapp-testcommon.md [hunk 1]
```diff
@@ -44,6 +44,14 @@ Currently we have the following managers:
...
 
+### Resource providers
+
+The `ServiceBusResourceProvider` is more complex than a *manager*, so we named it differently.
+
+It makes it easy to build a bunch of resources within the same Azure Service Bus namespace, and will automatically track and cleanup any resources created, when it is disposed.
+
+Queues and topics created using the resource provider, will be created using a combination of a given prefix and a random suffix. This is to ensure multiple runs of the same tests can run in parallel without interferring.
+
 ### Verify Service Bus messaging
```
Comment (dstsyst) [inline source/TestCommon/documents/functionapp-testcommon.md:56]: Add description of "IntegrationTestConfiguration"

File: source/TestCommon/documents/release-notes/release-notes.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # TestCommon Release notes
 
+## Version 1.0.0
+
+- Implemented `ServiceBusResourceProvider`. See [Resource providers](../functionapp-testcommon.md#resource-providers).
+- Implemented `IntegrationTestConfiguration` to support using Integration Test environment.
```
Comment (dstsyst) [inline source/TestCommon/documents/release-notes/release-notes.md:6]: Link to documentation of "IntegrationTestConfiguration" when ready

File: source/TestCommon/source/FunctionApp.TestCommon.Tests/Integration/ServiceBus/ResourceProvider/ServiceBusResourceProviderTests.cs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+using Energinet.DataHub.Core.FunctionApp.TestCommon.ServiceBus.ResourceProvider;
+using FluentAssertions;
+using Xunit;
+
+namespace Energinet.DataHub.Core.FunctionApp.TestCommon.Tests.Integration.ServiceBus.ResourceProvider
+{
+    // PoC on using identities to manage Azure resources (locally run tests as developer, on build agent run tests as SPN).
+    public class ServiceBusResourceProviderTests
+    {
+        public class UsingKeyVaultSecrets
```
Comment (dstsyst) [inline source/TestCommon/source/FunctionApp.TestCommon.Tests/Integration/ServiceBus/ResourceProvider/ServiceBusResourceProviderTests.cs:29]: Rename test class and create lots of tests

**Review by dstsyst — COMMENTED** [review]
Added a few comments

**Review by djorgensendk — DISMISSED** [review]

**Review by djorgensendk — APPROVED** [review]

File: source/TestCommon/documents/functionapp-testcommon.md [hunk 4]
```diff
@@ -24,6 +24,14 @@ The following only introduce the types supporting integration testing an Azure F
 
 > For a concrete implementation example take a look at the [Charges](https://github.com/Energinet-DataHub/geh-charges) repository/domain.
 
+### Integration Test environment
+
+The *Integration Test environment* is a resource group containing shareable Azure resources to support various integration test scenarious. E.g. this resource group contains a Azure Service Bus namespace, so we don't have to spent time creating one in our tests.
```
Comment (jonasdmoeller) [inline source/TestCommon/documents/functionapp-testcommon.md:29]:
```suggestion
The *Integration Test environment* is a resource group containing shareable Azure resources to support various integration test scenarios. E.g. this resource group contains a Azure Service Bus namespace, so we don't have to spent time creating one in our tests.
```

**Review by jonasdmoeller — APPROVED** [review]

**Review by jonasdmoeller — APPROVED** [review]

**Review by Renetnielsen — APPROVED** [review]

**Pull request merged**